src/input/keyboard_and_mouse.rs

```diff
@@ -51,17 +55,35 @@ impl KeyboardAndMouse {
     pub fn was_key_released(&self, key_code: keyboard::KeyCode) -> bool {
         self.released_keys.contains(&key_code)
     }
+
+    /// Returns true if the given button is currently pressed.
+    pub fn is_button_pressed(&self, button: mouse::Button) -> bool {
+        self.pressed_buttons.contains(&button)
+    }
+
+    /// Returns the positions of the mouse clicks during the last interaction.
+    ///
+    /// Clicks performed while the mouse cursor is not available are
+    /// automatically ignored.
+    pub fn button_clicks(&self, button: mouse::Button) -> &[Point] {
```

Comment:
I think we should probably rename it to `mouse_button_clicks`, as keyboards can have buttons too... Same with `is_mouse_button_pressed`. Although this feels a bit too verbose...

I am thinking that maybe we could split `KeyboardAndMouse` into two structs: `Keyboard` and `Mouse`, and use composition. Then it could be used like this (particles example):

```rust
fn interact(&mut self, input: &mut KeyboardAndMouse, window: &mut Window) {
    let mouse = input.mouse();

    self.gravity_centers[0] = mouse.cursor_position();
    self.gravity_centers.extend(mouse.clicks(mouse::Button::Left));

    let keyboard = input.keyboard();

    if keyboard.was_key_released(keyboard::KeyCode::I) {
        self.interpolate = !self.interpolate;
    }

    if keyboard.was_key_released(keyboard::KeyCode::F) {
        window.toggle_fullscreen();
    }
}
```

Both `Keyboard` and `Mouse` could implement `Input` too.

In any case, I am okay with just renaming these methods for now. We can work on this idea on another PR. That said, if you want to give it a shot here, feel free to do so!

Reply:
I think it is better to do it in a new PR after this. Do you want to create a issue for that?

Reply:
Yes, I created an issue to track it: https://github.com/hecrj/coffee/issues/68